### PR TITLE
chore: add keycloak for acc tests

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
This adds documentation for running a keycloak server that can be used to test SAML/OAuth and Browser based workflows. More work will need to be done to automate the process. Perhaps with https://github.com/compose-spec/compose-go to start up container automatically.

### CODE REVIEW

File: pkg/acceptance/keycloak/README.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+The following Terraform configuration can be used to authenticate using the OAuth access token
+
+```
```
Comment (sfc-gh-jcieslak) [inline pkg/acceptance/keycloak/README.md:114]: double ```

File: pkg/acceptance/keycloak/compose.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    command: 
+      - start-dev 
+      - --http-port=8080
+      - --import-realm
+    volumes:
+      - ./realm-export.json:/opt/keycloak/data/import/realm-export.json
+
+networks:
+  webnet:
```
Comment (sfc-gh-jcieslak) [inline pkg/acceptance/keycloak/compose.yml:18]: empty options ?

File: pkg/acceptance/keycloak/compose.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+volumes:
+  realm-data:
```
Comment (sfc-gh-jcieslak) [inline pkg/acceptance/keycloak/compose.yml:21]: empty options ?

File: pkg/acceptance/keycloak/realm-export.json [hunk 4]
```diff
@@ -0,0 +1,2450 @@
+{
```
Comment (sfc-gh-asawicki) [inline pkg/acceptance/keycloak/realm-export.json:1]: this is a really long json. I have a few questions:
- Do we need all of it?
- Isn't there any sensitive data leaked here?

**Comment by sfc-gh-asawicki** [comment]
> This adds documentation for running a keycloak server that can be used to test SAML/OAuth and Browser based workflows. More work will need to be done to automate the process. Perhaps with https://github.com/compose-spec/compose-go to start up container automatically.

I like it. I think we can merge it without adding any automation for it now. We can add it as a part of SNOW-999050.

I did not try to run it myself yet, but the description looks solid. I left just one question for the realm json.

**Pull request closed without merging**